### ISSUE TITLE
Typescript imports not showing function definitions?

### ISSUE THREAD

**Issue by dxu** [ms-jpq/coq_nvim#126]
Hi, I'm brand new to LSP and autocompletion, and relatively new to nvim, so please let me know if I am not asking the right questions or providing the right information. 

I can't seem to get autocompletion and type suggestions to work with `coq`, but it seems to work with `coc`. I'm not sure if it's an issue with the lsp server, or with my configuration. 

I'm trying to enable coq.vim for typescript files. I've installed tsserver, have installed nvim-lspconfig, and have a blank project running typescript. 

When I try to call on functions on imported objects, for example when typing:
```
import {debounce} from 'lodash'
debounce( // ...
```

The popup that shows up does not show any autocomplete information about the method. I tried separately setting up coc.vim for the first time, and it seemed to work. 

Here's a side by side screenshot of the two different instances and the steps i took. Left is coq, right is coc.

Typing out "debounce", I can actually see both docs:
![Screen Shot 2021-08-23 at 5 20 36 PM](https://user-images.githubusercontent.com/718423/130537055-dd9ebdbd-e6ae-4672-90df-5332c935f77f.png)

When selected, when I hit `(`, it shows docs on coc, but not with coq (which shows a generic function snippet:
![Screen Shot 2021-08-23 at 5 21 12 PM](https://user-images.githubusercontent.com/718423/130536076-8e078986-72e7-4f2a-8c8d-90148effec56.png)

I also tried by calling a method on an object from another import, like so:

```
import express from "express";
const app = express();
app.get(
```

In coc, it gives me type hints and the ability to complete, while coq doesn't show anything:
![Screen Shot 2021-08-23 at 5 21 36 PM](https://user-images.githubusercontent.com/718423/130536242-37ac7ff0-5b30-4585-882b-5a519323a8b0.png)

I'm a big fan of your work, and would love to support this project. Am I setting up something wrong with the LSP server? Is it a difference between coc-tsserver and tsserver?

**Comment by zeertzjq** [comment]
Maybe for the latter you'll need [ray-x/lsp_signature.nvim](https://github.com/ray-x/lsp_signature.nvim).

**Comment by ms-jpq** [comment]
This is because [Typescript does not actually implement LSP](https://github.com/microsoft/TypeScript/issues/39459), it provides completion using its own thing called `tsserver`, which predates `LSP`.

[coc forked `tsserver`](https://github.com/neoclide/coc-tsserver), and in theory should give results on par with `tsserver` (the gold standard).

---

The LSP you can install yourself: [typescript-language-server](https://github.com/typescript-language-server/typescript-language-server) actually is one of the worst LSP out there, as far as completion results go.

it's so bad that it actually helped me when developing `coq`, as like the benchmark "terrible" LSP server.

---

at this point, we can only wait for microsoft to finish their `LSP` adapter for `tsserver`. If you read their github issues, the typescript team have been working on this for a while now, but there is still no definitive release date.

**Comment by ms-jpq** [comment]
tldr, you are using different servers, and `coc`'s server is just better.

nothing I can do, because im have zero plans to submit patches to any of those LSP projects.
